server/src/main/java/org/opentosca/toscana/plugins/cloudformation/CloudFormationFileCreator.java

```diff
@@ -0,0 +1,173 @@
+package org.opentosca.toscana.plugins.cloudformation;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.opentosca.toscana.core.plugin.PluginFileAccess;
+import org.opentosca.toscana.plugins.scripts.BashScript;
+import org.opentosca.toscana.plugins.scripts.EnvironmentCheck;
+
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+
+/**
+ * Class for building scripts and copying files needed for deployment of cloudformation templates.
+ */
+public class CloudFormationFileCreator {
+    static final String CLI_COMMAND_CREATESTACK = "aws cloudformation deploy ";
+    static final String CLI_PARAM_STACKNAME = "--stack-name ";
+    static final String CLI_PARAM_TEMPLATEFILE = "--template-file ";
+    //TODO check if PARAMOVERRIDES are needed and get said parameters
+    // private static final String CLI_PARAM_PARAMOVERRIDES = "--parameter-overrides ";
+    static final String FILENAME_DEPLOY = "deploy";
+    static final String FILENAME_UPLOAD = "file-upload";
+    static final String TEMPLATE_PATH = "template.yaml ";
+    static final String CHANGE_TO_PARENT_DIRECTORY = "cd ..";
+    static final String RELATIVE_DIRECTORY_PREFIX = "../files/";
+
+    private final Logger logger;
+    private CloudFormationModule cfnModule;
+
+    /**
+     * Creates a <tt>CloudFormationFileCreator<tt> in order to build deployment scripts and copy files.
+     *
+     * @param cfnModule Module to get the necessary CloudFormation information
+     * @param logger    standard logger
+     */
+    public CloudFormationFileCreator(Logger logger, CloudFormationModule cfnModule) {
+        this.logger = logger;
+        this.cfnModule = cfnModule;
+    }
+
+    /**
+     * Copies all files that need to be uploaded to the target artifact.
+     */
+    public void copyFiles() {
+
+        ArrayList<String> filesToBeUploaded = cfnModule.getFilesToBeUploaded();
+
+        logger.debug("Checking if files need to be copied.");
+        if (!filesToBeUploaded.isEmpty()) {
+            logger.debug("Files to be copied found.");
+            logger.debug("Copying files to the target artifact.");
+            filesToBeUploaded.forEach((filePath) -> {
+                String targetPath = CloudFormationModule.FILEPATH_TARGET + filePath;
+                try {
+                    cfnModule.getFileAccess().copy(filePath, targetPath);
+                } catch (IOException e) {
+                    logger.error("Copying of files to the target artifact failed.");
+                    logger.error("See the stack trace for more info.");
+                    e.printStackTrace();
```

Comment:
you could add the error message to the log before.
In general, maybe you could consider to use "one-line-logs".

Reply:
It's probably best to just throw TransformationFailureExceptions here since these files are required in order for the transformation to be complete.